docs/CustomUIA.md

```diff
@@ -0,0 +1,115 @@
+<!-- Copyright (c) Microsoft Corporation. All rights reserved.
+     Licensed under the MIT Licence. -->
+
+## Custom UI Automation properties
+Axe Windows supports the inspection of [custom UI Automation properties](https://docs.microsoft.com/en-gb/windows/win32/winauto/uiauto-custompropertieseventscontrolpatterns). The custom properties for which Axe Windows should register are described in a user-supplied text file that is applied to all inspected UI Automation elements.
+
+### Configuration file format
+Custom UI Automation configuration data is specified in a [JSON](https://en.wikipedia.org/wiki/JSON#Syntax) formatted text file. The file consists of one object with the following possible attributes:
+
+#### metadata
+This attribute, if specified, can contain values to describe the file as determined by the user. It may be useful, for instance, to store version or authorship information. Example:
+
+```
+  "metadata": {
+    "version": "1.0",
+    "modified": "1970-01-01"
+  }
+```
+
+#### properties
+This attribute contains an array of objects, one per defined custom UI Automation property. Each property object contains the following fields:
+
+Attribute | Description
+--- | ---
+guid|The globally unique identifier of this property.
```

Comment:
It would be good for this to clarify the exact expected format of the GUID. If you don't have a reason to prefer any particular specific format, I'd reference [RFC4122](https://datatracker.ietf.org/doc/html/rfc4122#section-4.1) (this matches what [the latest draft of the JSON Schema spec](https://datatracker.ietf.org/doc/html/draft-handrews-json-schema-validation-02#section-7.3.5) does). If you do this, you'll want to drop the `{` and `}` characters in the example's GUIDs

Reply:
This is the format used both by NVDA and in the UIA docs for Excel and PowerPoint.

Reply:
As discussed, Newtonsoft can read guids with or without braces, but it writes guids without braces. Let's omit the braces in the spec.